arangod/Cluster/ResignShardLeadership.cpp

```diff
@@ -130,15 +119,31 @@ bool ResignShardLeadership::first() {
     // for now but we will not accept any replication operation from any
     // leader, until we have negotiated a deal with it. Then the actual
     // name of the leader will be set.
-    col->followers()->setTheLeader(LeaderNotYetKnownString);  // resign
-    res = methods->abort();                                   // unlock
+    for (auto const& s : shards) {
+      auto col = vocbase->lookupCollection(s);
+      if (col == nullptr) {
+        std::stringstream error;
+        error << "Failed to lookup local collection " << collection
```

Comment:
Probably we should include the shard name in the error message here as well, to have more detail should this error happen, e.g.
"Failed to lookup shard _s_ of local collection _collection_"